Lib/unittest/test/test_case.py

```diff
@@ -306,6 +306,21 @@ def test(self):
 
         Foo('test').run()
 
+    def test_deprecation_of_return_val_from_test(self):
+        # Issue 41322 - deprecate return of value!=None from a test
+        class Foo(unittest.TestCase):
+            def test(self):
+                return 1
+            def test2(self):
+                yield 1
```

Comment:
Why there are two test methods?

Reply:
The initial bug report was for generator function being used as a test case by accident, and later it was suggested to disallow all return values other than None. I thought it would be nice to cover the generator test func scenario specifically. But I don't mind removing it.

Reply:
Ah, I see it now. It is up to you. But if keep the generator test here, maybe add an asynchronous generator test for IsolatedAsyncioTestCase?